projects/storefrontapp-e2e-cypress/cypress/helpers/b2b/b2b-replenishment-order-details.ts

```diff
@@ -0,0 +1,129 @@
+import {
+  b2bProduct,
+  POWERTOOLS_BASESITE,
+} from '../../sample-data/b2b-checkout';
+import * as alerts from '../global-message';
+import { doPlaceOrder } from '../order-history';
+import {
+  replenishmentCancelDialogSelector,
+  replenishmentOrderHistorySelector,
+  visitReplenishmentHistory,
+} from './b2b-replenishment-order-history';
+
+export let replenishmentOrderDetailsUrl = (replenishmentOrderCode: string) =>
+  `/${POWERTOOLS_BASESITE}/en/USD/my-account/my-replenishment/${replenishmentOrderCode}`;
+
+export const orderOverviewSelector = 'cx-order-overview';
+export const cartListItemSelector = 'cx-cart-item-list';
+export const orderHistorySelector = 'cx-order-history';
+export const cancellationSelector = 'cx-replenishment-order-cancellation';
+
+export function waitForReplenishmentDetailsRequest(
+  replenishmentOrderCode: string
+) {
+  const replenishmentDetailsAlias = 'replenishmentDetailsAlias';
+
+  cy.server();
+
+  cy.route(
+    'GET',
+    `${Cypress.env('OCC_PREFIX')}/${Cypress.env(
+      'BASE_SITE'
+    )}/users/*/replenishmentOrders/${replenishmentOrderCode}*`
+  ).as(replenishmentDetailsAlias);
+
+  return `@${replenishmentDetailsAlias}`;
+}
+
+export function clickReplenishmentDetails(replenishmentOrderCode: string) {
+  const replenishmentDetails = waitForReplenishmentDetailsRequest(
+    replenishmentOrderCode
+  );
+
+  cy.get(
+    `${replenishmentOrderHistorySelector} .cx-replenishment-order-history-code a`
+  )
+    .click()
+    .should('have.attr', 'href')
+    .and('eq', replenishmentOrderDetailsUrl(replenishmentOrderCode));
+
+  cy.wait(replenishmentDetails).its('status').should('eq', 200);
+
+  return replenishmentDetails;
+}
+
+export function cancelReplenishmentDetails() {
+  doPlaceOrder(b2bProduct).then((orderData: any) => {
+    cy.waitForOrderToBePlacedRequest(
+      POWERTOOLS_BASESITE,
+      undefined,
+      orderData.body.replenishmentOrderCode
+    );
+
+    visitReplenishmentHistory();
+
+    const replenishmentDetails = clickReplenishmentDetails(
+      orderData.body.replenishmentOrderCode
+    );
+
+    cy.get(replenishmentDetails).then((xhr) => {
+      const body = xhr.response.body;
+
+      expect(body.replenishmentOrderCode).to.eq(
+        orderData.body.replenishmentOrderCode
+      );
+
+      expect(body.active).to.equal(true);
+    });
+
+    verifyOrderOverviewIsNotCancelled();
+
+    cy.get(`${cartListItemSelector} .cx-link`).should(
+      'contain',
+      orderData.body.entries[0].product.name
+    );
+
+    cy.get(orderHistorySelector).should('exist');
+
+    cancellationButtons(2);
+
+    cy.get(cancellationSelector)
+      .contains('Cancel')
+      .should('be.visible')
+      .click();
+    cy.get(`${replenishmentCancelDialogSelector}`)
+      .should('exist')
+      .then((_) => {
+        cy.get(`${replenishmentCancelDialogSelector} .btn-primary`).click();
```

Comment:
Please, add a wait for the cancellation alias here and assert on it additionally to the success message assertion

 cy.route(
      'PATCH',
      `${Cypress.env('OCC_PREFIX')}/${Cypress.env(
        'BASE_SITE'
      )}/users/*/replenishmentOrders/${orderData.body.replenishmentOrderCode}*`
    ).as('cancelAlias');

Reply:
😮 

It works for details page!!!!!!!! (just tried it, but when I tried to re-use in history the createRoute, it didn't work)

Reply:
Another note, re-using the route created in history does not work when cancelling in history
1) using a glob 'etc/etc/replenishement** and passing the requestMethod did not work 
2) even changing the route to 'etc/etc/replenishment/${code} when it's `PUT`
3) even with above and changing the alias did not work

HOWEVER! it works if I use the route created from the details page. I am very confused

Reply:
all in all, fixed! Thanks for the feedback